src/prop/proof_cnf_stream.cpp

```diff
@@ -571,6 +571,32 @@ void ProofCnfStream::convertPropagation(theory::TrustNode trn)
   d_psb.clear();
 }
 
+void ProofCnfStream::convertPropagationTrusted(Node exp, Node lit)
+{
+  // we need however to register the assumption as this might break the
+  // expansion when finalizing the proof (for trying to explain as a literal
+  // what is actually a clause). Since we have no proofs in theory engine,
+  // we cannot go via tte.getProven(). The explanation is decomposing
```

Comment:
Actually I'm confused, tte.getProven() is fine to call for any non-null TrustNode, regardless of whether proofs are enabled.

Reply:
the issue was not that I couldn't call it, but that `getProven` would give the same result as `getNode` when we are not producing proofs in the SMT engine.

Reply:
I think this would suggest there is a bug.  What TrustNodeKind are the TrustNode in question?

Reply:
I didn't check the invariants to see if it was a bug of expected behavior. Maybe we have to change [this line](https://github.com/CVC4/CVC4/blob/e6168da0663c73f17994b50f62a11799619b67ba/src/theory/theory_engine.cpp#L1775) to generate a trust node for prop explain?

Reply:
Yes, the line you reference looks wrong to me.

Reply:
OK, I'll take a look.